pkg/generate/app/pipeline.go

```diff
@@ -64,7 +64,7 @@ func (pb *pipelineBuilder) NewBuildPipeline(from string, resolvedMatch *Componen
 			return nil, fmt.Errorf("can't build %q: %v", from, err)
 		}
 		input = inputImage
-		if !input.AsImageStream {
+		if !input.AsImageStream && resolvedMatch.Value != "scratch" {
```

Comment:
Scratch has a specific meaning only to docker images.  I doubt we would use a special name for "scratch" on a non-Docker based layered builder, so in whatever followup we do to ensure "scratch" is correctly handled by all the other tooling, we need to formalize the transformation of "Docker scratch reference" -> "generic concept meaning on top of nothing".